Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/GameFileCache.java

```diff
@@ -28,67 +22,89 @@ public GameFileCache(String path)
   @Override
   public native void finalize();
 
-  public static void addGameFolder(String path, Context context)
+  public static void addGameFolder(String path)
   {
-    SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(context);
-    Set<String> folderPaths = preferences.getStringSet(GAME_FOLDER_PATHS_PREFERENCE, EMPTY_SET);
+    File dolphinFile = SettingsFile.getSettingsFile(Settings.FILE_DOLPHIN);
+    IniFile dolphinIni = new IniFile(dolphinFile);
+    LinkedHashSet<String> pathSet = getPathSet(false);
+    int totalISOPaths =
+            dolphinIni.getInt(Settings.SECTION_INI_GENERAL, SettingsFile.KEY_ISO_PATHS, 0);
 
-    if (folderPaths == null)
+    if (!pathSet.contains(path))
     {
-      return;
+      dolphinIni.setInt(Settings.SECTION_INI_GENERAL, SettingsFile.KEY_ISO_PATHS,
+              totalISOPaths + 1);
+      dolphinIni.setString(Settings.SECTION_INI_GENERAL, SettingsFile.KEY_ISO_PATH_BASE +
+              totalISOPaths, path);
+      dolphinIni.save(dolphinFile);
```

Comment:
Please call `NativeLibrary.ReloadConfig()` after saving here, otherwise the core might steamroll over the changes you just did once these settings have been migrated to the new config system.

Reply:
Calling `ReloadConfig` would also make it possible for us to change `GameFileCache.update` to read from the in-memory config instead of needing to pass an array with paths. (I can do that later in a separate PR.)